src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -45,41 +46,48 @@ internal abstract partial class MethodExtractor
         protected abstract SyntaxToken GetMethodNameAtInvocation(IEnumerable<SyntaxNodeOrToken> methodNames);
         protected abstract ImmutableArray<AbstractFormattingRule> GetCustomFormattingRules(Document document);
 
-        protected abstract Task<OperationStatus> CheckTypeAsync(Document document, SyntaxNode contextNode, Location location, ITypeSymbol type, CancellationToken cancellationToken);
+        protected abstract OperationStatus CheckType(SemanticModel semanticModel, SyntaxNode contextNode, Location location, ITypeSymbol type);
 
-        protected abstract Task<(Document document, SyntaxToken methodName, SyntaxNode methodDefinition)> InsertNewLineBeforeLocalFunctionIfNecessaryAsync(Document document, SyntaxToken methodName, SyntaxNode methodDefinition, CancellationToken cancellationToken);
+        protected abstract Task<(Document document, SyntaxToken methodName)> InsertNewLineBeforeLocalFunctionIfNecessaryAsync(Document document, SyntaxToken methodName, SyntaxNode methodDefinition, CancellationToken cancellationToken);
 
         public async Task<ExtractMethodResult> ExtractMethodAsync(CancellationToken cancellationToken)
         {
             var operationStatus = OriginalSelectionResult.Status;
 
-            var analyzeResult = await AnalyzeAsync(OriginalSelectionResult, LocalFunction, cancellationToken).ConfigureAwait(false);
+            var originalSemanticDocument = OriginalSelectionResult.SemanticDocument;
+            var analyzeResult = Analyze(OriginalSelectionResult, LocalFunction, cancellationToken);
             cancellationToken.ThrowIfCancellationRequested();
 
-            operationStatus = await CheckVariableTypesAsync(analyzeResult.Status.With(operationStatus), analyzeResult, cancellationToken).ConfigureAwait(false);
+            operationStatus = CheckVariableTypes(analyzeResult.Status.With(operationStatus), analyzeResult, cancellationToken);
             if (operationStatus.Failed())
                 return new FailedExtractMethodResult(operationStatus);
```

Comment:
this part is now synchronous and works without forking.  in the followup PR this is the point that we will decide to show the code action.

what follows will be moved into the code-action-commit portion.